src/main/java/org/sonarlint/intellij/actions/AbstractSonarAction.java

```diff
@@ -39,15 +39,29 @@ public AbstractSonarAction(@Nullable String text, @Nullable String description,
   public void update(AnActionEvent e) {
     Project p = e.getProject();
 
-    e.getPresentation().setVisible(true);
+    if (isVisible(e.getPlace())) {
+      e.getPresentation().setVisible(true);
+    } else {
+      e.getPresentation().setVisible(false);
+      e.getPresentation().setEnabled(false);
+      return;
+    }
 
     if (p == null || !p.isInitialized() || p.isDisposed()) {
       e.getPresentation().setEnabled(false);
     } else {
       SonarLintStatus status = SonarLintStatus.get(p);
-      e.getPresentation().setEnabled(isEnabled(p, status));
+      e.getPresentation().setEnabled(isEnabled(e, p, status));
     }
   }
 
-  protected abstract boolean isEnabled(Project project, SonarLintStatus status);
+  /**
+   * Whether the action should be visible in a place.
+   * Examples: MainMenu, ProjectViewPopup, GoToAction
+   */
+  protected boolean isVisible(String place) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "place". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172)